docs/DeveloperGuide.md

```diff
@@ -1,34 +1,920 @@
+<br/>
 
-## Non-Functional Requirements
+## 2. Setting Up
+### Setting up the project in your computer
 
-{Give non-functional requirements}
+Ensure that you have the following installed: 
+*   JDK 11.
+*   IntelliJ IDE (highly recommended).
 
-## Glossary
+Firstly, **fork** this repo and **clone** a copy into your computer.
 
-* *glossary item* - Definition
+If you plan to use Intellij IDEA: 
+1. **Ensure IntelliJ is configured to use JDK 11**.
+    1. Click on `Configure` > `Structure for New Projects` > `Project Settings` > `Project`, 
+       and ensure the `Project SDK` is using **JDK 11**.
+2. **Import the project as a Gradle project**.
+    1. Click on `Import Project` and locate the `build.gradle` file and select it. Click `OK`
+    2. If asked, choose to `Open as Project` (not `Open as File`).
+    3. Click `OK` to accept the default settings but do ensure that the selected version of `Gradle JVM` matches the JDK being used for the project.
+    4. The import process could take a few minutes to finish.
+3. **Verify the setup**: 
+    1. Run the `seedu.anichan.Main` and try a few commands. 
+    2. You may want to refer to our [User Guide](UserGuide.md) for the list of commands.
 
-## Instructions for manual testing
+----
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+### Before writing code
+1. **Configuring the coding style**
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;If using IDEA, follow this guide [IDEA: Configuring the code style](https://se-education.org/guides/tutorials/intellijCodeStyle.html) to setup IDEA’s coding style to match ours.
+    
+2. **Set up CI**
+    
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;There is no set up required as the project comes with a GitHub Actions config files, located in `.github/workflows` folder.
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;When GitHub detects these files, it will run the CI for the project automatically at each push to the master branch or to any PR.
+    
+3. **Learn the design**
+    
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;When you are ready to start writing codes, we recommended that you have a look at AniChan's overall design by reading about it at [AniChan's architecture](DeveloperGuide.md#31-architecture).
+    
+<br/>
+
+## 3. Design 
+
+The following section describes the architecture design of **Anichan**. This section starts off by looking at the overall architecture design in a general view, before going into the specific implementation details of the individual features.
+
+<br/>
+
+### 3.1 Architecture
+This section will help provide insight to the general overview of Anichan’s architecture.
+
+<br/>
+
+![Architectural Diagram](images/Architectural-Design.png)
+
+*Figure 1: Architecture Design Diagram*
+
+> :bulb: The images used are stored in the directory: `images/`. If you wish to update a diagram you may replace the images in this folder.
```

Comment:
It might be good to indicate what the light bulb image means at the beginning of the DG to help the user understand the logos and images used in this document!